state/src/extension.ts

```diff
@@ -0,0 +1,250 @@
+import {EditorState, StateField} from "./state"
+
+type A<T> = ReadonlyArray<T>
+
+// Priorities for overriding the ordering of extensions and behaviors.
+export enum Priority { fallback = -1, base = 0, extend = 1, override = 2 }
+
+const noPriority = -2 as Priority
+
+const noDefault = {} as any
+
+const none = [] as any
+
+// A behavior is a type of value that can be associated with an editor
+// state. It is used to configure the state, for example by
+// associating helper functions with it (see `Behavior.indentation`)
+// or configuring the way it behaves (see
+// `Behavior.allowMultipleSelections`).
+export class Behavior<Value> {
+  private constructor(public unique: boolean) {}
+
+  static define<Value>({unique = false}: {unique?: boolean} = {}) {
+    return new Behavior<Value>(unique)
+  }
+
+  use(value: Value, priority: Priority = noPriority): Extender {
+    return new Extender(null, this, value, priority)
+  }
+
+  // The the values for this behavior.
+  get(state: EditorState): Value[] {
+    return state.config.behavior.get(this)
+  }
+
+  // Get the value of a unique behavior, or a default value when it
+  // isn't available in this state.
+  getSingle<Default = undefined>(state: EditorState, defaultValue: Default): Value | Default {
+    if (!this.unique) throw new Error("Can only call getSingle on a Behavior with unique=true")
+    let all = this.get(state)
```

Comment:
@marijnh It's a simple suggestion.
Wouldn't it be better to use a const? This is a little more explicit. Notifying you that it hasn't changed.

Reply:
You could, but my style is to use `let` for local variables, since thinking about whether to use `const` or `let` (and adjusting them when you do need to mutate after all) seems more work than justified by the added clarity.

Reply:
@marijnh I wanted to know why you used 'let'. 
Thank you for your answer.